application/dash_application/views/bcl2fastq.py

```diff
@@ -158,41 +176,23 @@ def update_layout(run_alias):
             functions update_known_index_bar, update_unknown_index_bar,
             update_pie_chart's data value, and update_pie_chart's fraction value
         """
+        known_run = known_data_table[known_data_table[bcl2barcode_col.Run] == run_alias]
+        known_run = known_run[~known_run[PINERY_COL.SampleProvenanceID].isna()]
+        known_run = known_run.drop_duplicates([PINERY_COL.SampleProvenanceID, bcl2barcode_col.Lane])
```

Comment:
I don't think this is right. I think you have to group and sum the duplicates. If it duplicated, then there will be multiple barcodes, which are unique entities from the bcl2fastq perspective.

Reply:
This line is only considering the SampleProvenanceID and the Lane number for identifying whether a row is a duplicate. As long as those are duplicated they'll be dropped. Should I also be considering the indices as criteria so they don't get dropped?

Reply:
I'm not sure that makes sense either. I think dropping is the wrong operation. Suppose you have a 10X sample `foo` in lane 1 with barcode `SI-GA-H01`, after joining:

| foo | 1 | SI-GA-H01 | AAAA | 32 |
| foo | 1 | SI-GA-H01 | AATT | 30 |
| foo | 1 | SI-GA-H01 | AACC | 33 |
| foo | 1 | SI-GA-H01 | AAGG | 31 |

You want to end up with output that either has 4 stacked bars for `foo`, so, no dropping at all, or a single bar whose height is 32+30+33+31. Does that make sense?

Reply:
Right. I guess I'm confused, then, as to why the original version of this report did it: https://github.com/oicr-gsi/dashi/blob/5ac70e6e53780226402615ed3dbec8829a1a5670/application/dash_application/views/bcl2fastq.py#L165

Reply:
Probably because it was getting duplicate records from the messy data it was ingesting.

Reply:
Updated